source/docs/software/advanced-controls/introduction/approaches-to-ctrl-sys-design.rst

```diff
@@ -0,0 +1,70 @@
+Approaches to Designing Control Systems
+=======================================
+
+.. note:: This article includes sections of `Controls Engineering in FRC <https://file.tavsys.net/control/controls-engineering-in-frc.pdf>`__ by Tyler Veness with permission.
+
+When designing a control system for your robot, there are a number of different approaches to take. They range from very simple approaches, to advanced and complex ones. Each has *tradeoffs*. Some will work better than others in different situations, some require more mathematical analysis than others.
+
+Teams should prioritize picking the easiest strategy which enables success on the field. However, as you do experiments, keep in mind there is almost always a "next-step" to take to improve your field performance.
+
+Feedforward vs. Feedback
+------------------------
+
+There are two broad categories of strategies in control system design:
+
+**Feedforward** refers to the class of algorithms which incorporate knowledge of how the mechanism under control is *expected* to operate. Using this "model" of operation, the control input is chosen to make the mechanism get close to where it should be.
+
+**Feedback** refers to the class of algorithms which use sensors to *measure* what a mechanism is doing, and issue corrective commands to move a mechanism from where it actually is, to where you want it to be.
+
+They are not mutually exclusive - many techniques will use these concepts simultaneously.
+
+Start with Feedforward
+-----------------------
+
+Because most FRC mechanisms can be manufactured to be have well-defined behavior and can be easily modeled, it's generally encouraged to look into feedforward techniques first.
```

Comment:
At the risk of starting a holy war, I'm not sure I agree with this. A pure feedforward control scheme won't work with unmodeled disturbances, which there are many. If I was going to suggest either feedforward or feedback control, I'd have to say feedback control would be better for the vast majority of teams. Obviously feedforward + feedback is the best.

Reply:
It depends critically on what type of system you are trying to control.  See my post above for a more thorough discussion.